### ISSUE TITLE
some buildings obstruct exit of units from factories

### ISSUE THREAD

**Issue by pandaro** [eronoobos/BABAR-The-Shardifant#70]
there is the case that a large amount of buildings are builded next to the reserved area of factories exits,
well, when the  exit area is ended some buildings are builded in front of factory, obstructing in fact the exit.
i try to draw the problem : 
x = building
f = factory
a= area of respect
                     xxffffffffxx
                     xxffffffffxx
                     xxaaaaxx
                     xxaaaaxx
                     xxaaaaxx
                     xxaaaaxx
                     xxxxxxxx
                     xxxxxxxx

**Comment by eronoobos** [comment]
Yup. It would be simple to make the lanes longer, but that would make places to build factories harder to find.

**Comment by pandaro** [comment]
we can place factories and after extend area of respect?

**Comment by pandaro** [comment]
pre draw way on map:
when maphandler is init, maybe we can preserve a specie of way(street) on map, a grid of areapreserved from buildings. Each street is large like a big unit, and cant override mex and geo spot.
the number of this line is variable on the dimension of map and the number too. this line can be vertical and horizontal.

**Comment by eronoobos** [comment]
Building on that idea: I can use the pathfinder to get a path to the enemy base or maybe the furthest mex in the mobility network. The reserved exit street you propose could then be drawn along this path. It wouldn't have to go the entire way, just long enough to get out of range of any potential construction even in the late game (1500 radius or so?).

**Comment by eronoobos** [comment]
That could also be a way to determine which way the factory should face.

All this would mean needing a new way to find a position for the factory.
